packages/api/src/routes/medical/internal-cq.ts

```diff
@@ -144,9 +161,180 @@ router.post(
   "/directory/organization",
   requestLogger,
   asyncHandler(async (req: Request, res: Response) => {
+    const cxId = getUUIDFrom("query", req, "cxId").orFail();
     const body = req.body;
     const orgDetails = cqOrgDetailsOrgBizRequiredSchema.parse(body);
-    await createOrUpdateCQOrganization(orgDetails);
+    await createOrUpdateCQOrganization(cxId, orgDetails);
+
+    return res.sendStatus(httpStatus.OK);
+  })
+);
+
+/**
+ * GET /internal/carequality/directory/organization/v2/:oid
+ *
+ * Retrieves the organization with the specified OID from the Carequality Directory.
+ * @param req.params.oid The OID of the organization to retrieve.
+ * @returns Returns the organization with the specified OID.
+ */
+router.get(
+  "/directory/organization/v2/:oid",
```

Comment:
These endpoints are practically the same - can we update to either pass a param to signify whether its a org or facility or extract all the code thats the same and call it as a function